app/components/PlatformCards/PoapPlatform.tsx

```diff
@@ -0,0 +1,143 @@
+
+// Each platform is recognised by its ID
+const platformId: PLATFORM_ID = "POAP";
+
+const iamUrl = process.env.NEXT_PUBLIC_PASSPORT_IAM_URL || "";
+
+export default function PoapPlatform(): JSX.Element {
+  const { address, signer } = useContext(UserContext);
+  const { handleAddStamps, allProvidersState } = useContext(CeramicContext);
+  const [isLoading, setLoading] = useState(false);
+  const [canSubmit, setCanSubmit] = useState(false);
+
+  // find all providerIds
+  const providerIds =
+    STAMP_PROVIDERS["POAP"]?.reduce((all, stamp) => {
+      return all.concat(stamp.providers?.map((provider) => provider.name as PROVIDER_ID));
+    }, [] as PROVIDER_ID[]) || [];
+
+  // SelectedProviders will be passed in to the sidebar to be filled there...
+  const [verifiedProviders, setVerifiedProviders] = useState<PROVIDER_ID[]>(
+    providerIds.filter((providerId) => typeof allProvidersState[providerId]?.stamp?.credential !== "undefined")
+  );
+  // SelectedProviders will be passed in to the sidebar to be filled there...
+  const [selectedProviders, setSelectedProviders] = useState<PROVIDER_ID[]>([...verifiedProviders]);
+
+  // any time we change selection state...
+  useEffect(() => {
+    if (selectedProviders.length !== verifiedProviders.length) {
+      setCanSubmit(true);
+    }
+    if (selectedProviders.length === 0) {
+      setCanSubmit(false);
+    }
+  }, [selectedProviders, verifiedProviders]);
+
+  // --- Chakra functions
+  const toast = useToast();
+
+  // fetch VCs from IAM server
+  const handleFetchCredential = (): void => {
+    datadogLogs.logger.info("Saving Stamp", { platform: platformId });
+    fetchVerifiableCredential(
+      iamUrl,
+      {
+        type: platformId,
+        types: selectedProviders,
+        version: "0.0.0",
+        address: address || "",
+        proofs: {},
+      },
+      signer as { signMessage: (message: string) => Promise<string> }
+    )
+      .then(async (verified: VerifiableCredentialRecord): Promise<void> => {
+        // because we provided a types array in the params we expect to receive a
+        // credentials array in the response...
+        const vcs =
+          verified.credentials
+            ?.map((cred: CredentialResponseBody): Stamp | undefined => {
+              if (!cred.error) {
+                // add each of the requested/received stamps to the passport...
+                return {
+                  provider: cred.record?.type as PROVIDER_ID,
+                  credential: cred.credential as VerifiableCredential,
+                };
+              }
+            })
+            .filter((v: Stamp | undefined) => v) || [];
+        // Add all the stamps to the passport at once
+        await handleAddStamps(vcs as Stamp[]);
+        datadogLogs.logger.info("Successfully saved Stamp", { platform: platformId });
+        const verifiedProviders = providerIds.filter(
+          (providerId) => typeof allProvidersState[providerId]?.stamp?.credential !== "undefined"
```

Comment:
Can we update this to:
```suggestion
            !!vcs.find((vc: Stamp | undefined) => vc?.credential?.credentialSubject?.provider === providerId)
```
It will make sure that update happens in the same tick (so that the user doesn't have to close the sidebar and reopen it)